BigBaseV2/src/hooks/protections/received_clone_sync.cpp

```diff
@@ -25,38 +29,77 @@ namespace big {
 		break;
 	*/
 	
-
+	CBaseModelInfo* get_model_data(rage::joaat_t hash) {
+		auto modelTble = g_pointers->m_model_table;
+		for (auto i = modelTble->m_lookup_table[hash % modelTble->m_lookup_key]; i; i = i->m_next)
+		{
+			if (i->m_hash == hash)
+			{
+				if (const auto model = modelTble->m_data[i->m_idx]; model)
+				{
+					return model;
+				}
+			}
+		}
+		return nullptr;
+	}
 	int64_t hooks::received_clone_sync(CNetworkObjectMgr* mgr, CNetGamePlayer* src, CNetGamePlayer* dst, uint16_t sync_type, uint16_t obj_id, rage::datBitBuffer* buffer, uint16_t unk, uint32_t timestamp) {
-
-		auto sync_tree = g_pointers->m_get_sync_tree_for_type(mgr, sync_type);
-		auto tree_name = g_pointers->m_get_sync_type_info(sync_type, 0);
-		auto net_obj = g_pointers->m_get_net_object(mgr, obj_id, true);
-		bool invalidsync = false;
-
-		if(!net_obj) net_obj = g_pointers->m_get_net_object_for_player(mgr, obj_id, src, true);
-
-		if (!net_obj) return 2;
-
-		if (!sync_tree || sync_type < 0 || sync_type > 14) invalidsync = true;
-
-		if (net_obj->m_object_type != sync_type) invalidsync = true;
-
-		//TO BE ADDED
-		//Node specific entity type checks
-
-
-		if (invalidsync) {
-			if (g->notifications.invalid_sync.log)
-				LOG(WARNING) << "Invalid sync: " << "Type: " << sync_type << " Tree name: " << tree_name << " From: " << src->get_name();
-			if (g->notifications.invalid_sync.notify)
-				g_notification_service->push_warning(fmt::format("Invalid Sync from {}", src->get_name()), fmt::format("Type {} in sync tree {}", sync_type, tree_name));
-
-			return 2;
+		if (auto sync_tree = g_pointers->m_get_sync_tree_for_type(mgr, sync_type); sync_tree && NETWORK::NETWORK_IS_SESSION_ACTIVE())
+		{
+			if (auto net_obj = g_pointers->m_get_net_object(mgr, obj_id, true); net_obj)
+			{
+				auto tree_name = g_pointers->m_get_sync_type_info(sync_type, 0);
+				bool invalidsync = false;
+
+				if (sync_type < 0 || sync_type > 14)
```

Comment:
sync_type? Could you make an enum of these?

Reply:
> sync_type? Could you make an enum of these?

Blame Dayib, not me.

Reply:
I'll add the enum